addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2187,9 +2187,9 @@ const ListUserValueWidget = UserValueWidget.extend({
         const values = [...this.listTable.querySelectorAll('.o_we_list_record_name input')].map(el => {
             const id = this.isCustom ? el.value : el.name;
             return Object.assign({
-                id: /^-?[0-9]{1,15}$/.test(id) ? parseInt(id) : id,
-                name: el.value,
-                display_name: el.value,
+                id: /^-?[0-9]{1,15}$/.test(id) ? parseInt(id) : id.trim().replace(/\s+/g, " "),
+                name: el.value.trim().replace(/\s+/g, " "),
+                display_name: el.value.trim().replace(/\s+/g, " "),
```

Comment:
I was a bit worried that only trimming the `we-list`'s values in such a way would cause mismatch if those values are linked to things stored in DB, but it seems OK everywhere I could test. (Mostly because m2m widgets flag the `we-list` as `notEditable`, I think.)

That said, trimming the id should be done everywhere (because `"10 "` on a 1st loop will give a string, which will be trimmed, so on the 2nd loop it will give a number. And I don't think the 1st/2nd loops should be different). So I suggest:

```js
const trimmed = (str) => str.trim().replace(/\s+/g, " ");
id = trimmed(id);
return Object.assign({
    id: /^-?[0-9]{1,15}$/.test(id) ? parseInt(id) : id,
    name: trimmed(el.value),
    display_name: trimmed(el.value),
```

Reply:
By the way, shouldn't the `id` on line 2200 just below also be trimmed?

`                return /^-?[0-9]{1,15}$/.test(id) ? parseInt(id) : id;`

Reply:
I don't think we should trim it again, as it will always be the trimmed value set in the previous computation. I didn't find any possibility of needing a trim.

Reply:
I'm not sure we're talking about the same thing.
Line 2200 is within a distinct loop on the DOM elements, to check which ones are selected. Not trimming the `id` there means the `includes` check that happens afterwards (`this.selected.includes(v.id)`) won't be `true` if the `id` needs trimming (because `v.id` will be trimmed, but the id in `this.selected` won't).

In addition to that, as I had a look at your last commit, note that you don't need the `trimmed(id)` on line 2191: it's already done.

Reply:
Got it! Previously, we were considering the ID as `input.name` to check the selected checkbox, which had already been trimmed. However, with the recent merge, we are now using `input.value` for the custom fields. So yes, it's indeed correct.
Thanks for pointing that out ; )